pipeline/outputs/s3.md

```diff
@@ -25,9 +25,9 @@ Records are stored in files in S3 as newline delimited JSON.
 | upload\_chunk\_size | The size of each 'part' for multipart uploads. Max: 50M | 5,242,880 bytes |
 | upload\_timeout | Whenever this amount of time has elapsed, Fluent Bit will complete an upload and create a new file in S3. For example, set this value to 60m and you will get a new file every hour. | 10m |
 | store\_dir | Directory to locally buffer data before sending. When multipart uploads are used, data will only be buffered until the `upload_chunk_size` is reached. | /tmp/fluent-bit/s3 |
-| s3\_key\_format | Format string for keys in S3. This option supports strftime time formatters and a syntax for selecting parts of the Fluent log tag using a syntax inspired by the rewrite\_tag filter. Add $TAG in the format string to insert the full log tag; add $TAG\[0\] to insert the first part of the tag in the s3 key. The tag is split into “parts” using the characters specified with the `s3_key_format_tag_delimiters` option. See the in depth examples and tutorial in the documentation. | /fluent-bit-logs/$TAG/%Y/%m/%d/%H/%M/%S |
+| s3\_key\_format | Format string for keys in S3. This option supports a UUID, strftime time formatters, a syntax for selecting parts of the Fluent log tag using a syntax inspired by the rewrite\_tag filter and key suffix. Add $UUID in the format string to insert a random string. Add $TAG in the format string to insert the full log tag; add $TAG\[0\] to insert the first part of the tag in the s3 key. The tag is split into “parts” using the characters specified with the `s3_key_format_tag_delimiters` option. Add extension directly after the last piece of the format string to insert key suffix. If you want to specify a key suffix and you are in `use_put_object` mode, you must specify $UUID as well. More explanations can be found in `use_put_object` option. See the in depth examples and tutorial in the documentation. | /fluent-bit-logs/$TAG/%Y/%m/%d/%H/%M/%S |
 | s3\_key\_format\_tag\_delimiters | A series of characters which will be used to split the tag into 'parts' for use with the s3\_key\_format option. See the in depth examples and tutorial in the documentation. | . |
-| use\_put\_object | Use the S3 PutObject API, instead of the multipart upload API. | false |
+| use\_put\_object | Use the S3 PutObject API, instead of the multipart upload API. When this option is on, key extension is only available when $UUID is specified. If $UUID is not included, a random string will be appended at the end of the format string and the key extension cannot be customized in this case. | false |
```

Comment:
Be very clear, "when $UUID is specified in s3_key_format"

Reply:
Thanks